src/lib.rs

```diff
@@ -35,3 +38,38 @@ pub fn default_cfg() -> Vec<PathBuf> {
 
     cfgs
 }
+
+#[derive(Clone, Debug)]
+pub enum PortArg {
+    Number(u16),
+    Address(String),
```

Comment:
I think this would be better as a SocketAddr and the from_str should make use of the SocketAddr parsing - which seems to be used in TcpServer::new, so may as well use it here.

This also probably doesn't need to be an enum, just a wrapper type around SocketAddr.